rfd/0002/README.mdx

```diff
@@ -0,0 +1,198 @@
+—--
+authors: Nicolas Joseph <nicolas@nicolasjoseph.com>
+title: Ecosystem Skeleton Architecture
+state: discussion
+—--
+
+## Framework Problem Statement:
+
+We are building an opinionated framework to enabled engineers that are building data infrastructure and pipeline to focus on the business logic and not the infrastructure. We aim at providing them with the same user experience as a front end engineer building a web application. We want to provide a framework that is easy to use, easy to maintain, easy to extend and most importantly, gives the user feedback on a code iteration immediately.
+
+## Principles
+
+- **Local first**: We optimize for the user experience first and we will figure out how to deploy to production later.
+- **Easy to use**: We want to provide a framework that is easy to use, easy to maintain, easy to extend and most importantly, gives the user feedback on a code iteration immediately.
+- **Easy to test**: We want to provide a framework that has built in unit testing from the get go.
+- **Performance**: We want to provide a framework that is fast and can scale to large data sets and volumes for real life use cases.
+- **Extensible**: We want to provide a framework that is easy to extend and that can be used to build a wide range of data applications.
+
+## Architecture
+
+### Overview
+
+- Data Ingest SDK: The data ingest SDK is a simple library that can be embedded in the user's application to send data to an igloo deployment. It is typed with the User's Business Objects to ensure that the data sent to igloo is valid.
+- Igloo Server: This is a component that abstracts away several parts of the architecture. It contains a web server that will ingest the data, a streaming engine that will store the data in streams and an OLAP Database that will store the data in a columnar format. It also runs all the code that is supplied by the user to transform the data.
+- Igloo CLI: This is a command line interface that will allow the user to interact with the igloo server. It will allow the user to create a new project, run the project locally and deploy the project to a cloud provider.
+- Data Consumption SDK: This SDK is typed with the user's data and allows the user to query the data from the Igloo Server. It will also allow the user to subscribe to data changes and receive updates in real time.
```

Comment:
Are you going to include governance in the consumption and or server?

Reply:
🤔 another good callout, I have not considered governance yet, and how it lays out. 

Since we can't trust the client I would say both. 
* The Security / Access aspect needs to be handled on the server side. 
* The SDK / Client should be type safe from a quality of life perspective

In terms of data quality, I am thinking we could check the data at ingestion against the data schema / contract.

Reply:
Governance probably requires its own Discussion. I added it to the open questions